source/camera/CullingManager.cpp

```diff
@@ -0,0 +1,292 @@
+        return;
+
+    for (int jj = 0; jj < gm->getMapSizeY() ; ++jj)
+    {
+        for (int ii = 0; ii < gm->getMapSizeX(); ++ii)
+        {
+            gm->getTile(ii,jj)->show();
+        }
+    }
+}
+
+void CullingManager::newBashAndSplashTiles(int64_t mode){
+    int64_t xxLeftOld = mOldWalk.getTopLeftVertex().x;
+    int64_t xxRightOld= mOldWalk.getTopRightVertex().x;
+    int64_t xxLeft = mWalk.getTopLeftVertex().x;
+    int64_t xxRight= mWalk.getTopRightVertex().x;
+    int64_t xxp, yyp;
+    std::stringstream ss;
+    int64_t bb = (( std::min(mWalk.getBottomLeftVertex().y , mOldWalk.getBottomRightVertex().y) >> mPrecisionDigits) - 2) << mPrecisionDigits;
+
+    for (int64_t yy = ((std::max(mWalk.getTopLeftVertex().y , mOldWalk.getTopRightVertex().y  ) >> mPrecisionDigits) + 2) << mPrecisionDigits;  yy >= bb; yy -= Unit)
+    {
+        mOldWalk.notifyOnMoveDown(yy);
+        mWalk.notifyOnMoveDown(yy);
+        xxLeft = mWalk.getCurrentXLeft(yy);
+        xxLeftOld = mOldWalk.getCurrentXLeft(yy);
+        xxRight = mWalk.getCurrentXRight(yy);
+        xxRightOld = mOldWalk.getCurrentXRight(yy);
+   
+        // if(mDebug)
+        // {
+        //     ss << std::endl;
+        //     ss << "xxLeftOld, xxRightOld: " << double(xxLeftOld)/Unit << " " << double(xxRightOld)/Unit << " " ;
+        //     LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL,__FILE__,__LINE__, ss.str());
+        //     ss.str("");
+        //     ss << "xxLeft " << (xxLeft  >> mPrecisionDigits) << "xxRight " << (xxRight  >> mPrecisionDigits );
+        //     LogManager::getSingleton().logMessage(LogMessageLevel::NORMAL, __FILE__,__LINE__, ss.str());
+        //     ss.str("");
+        // }
+    
+        int64_t mm = ((std::min(xxLeft, xxLeftOld) >> mPrecisionDigits) << mPrecisionDigits) ;
+        if(std::min(xxLeft, xxLeftOld) < std::max(xxRight,xxRightOld) )
+        {
+            for (int64_t xx = mm ; xx <= std::max(xxRight,xxRightOld) ; xx+= Unit)
+            {
+                bool bash = (xx >= xxLeftOld && xx <= xxRightOld && (yy >= mOldWalk.getBottomLeftVertex().y) && yy <= mOldWalk.getTopLeftVertex().y);
+                bool splash = (xx >= xxLeft && xx <= xxRight && (yy >= mWalk.getBottomLeftVertex().y) && yy <= mWalk.getTopLeftVertex().y);
+
+                if (mDebug)
+                {
+                    ss<< " x " <<  (xx >> mPrecisionDigits )<< " y " << (yy >> mPrecisionDigits ) << " " << (splash && (mode & SHOW)) << (bash && (mode & HIDE)) << std::endl;
+            
+                }
+                xxp = ((xx >>mPrecisionDigits));
+                yyp = ((yy >>mPrecisionDigits));
+                GameMap* gm = mCm->mGameMap;
+                if(bash && splash && (mode & HIDE) && (mode & SHOW))
+                {
+                    // Nothing
+                }
+
+                else if (gm && bash && (mode & HIDE) && xxp>=0 && yyp>= 0 && xxp<mCm->mGameMap->getMapSizeX() && yyp < mCm->mGameMap->getMapSizeY())
```

Comment:
Can the gamemap really be null ?
Moreover, the check you are doing here is also done in GameMap::getTile. It would be better to check if the tile returned by GameMap::getTile is null.